### PR TITLE
fix: On user logout clear saved project groups on the Dashboard

### LINKED ISSUE #578: On user logout clear saved project groups on the Dashboard
**Describe the bug**
User can see cached project groups data:
image

**To Reproduce**

1. Sign in with user/password (Password sign-in)
2. Notice that project group dropdown on Dashboard is populated
3. Logout
4. Stay on the same page without refresh and sign in with Google auth user that has associated email domain that is not allowed (allowed_email_domains)
5. Notice project groups are displayed still because they were cached in the app (if you refresh the page on step 4 - it will not display any groups)

**Expected behavior**
On user logout - the project groups cache is cleared

**Additional context**

### PR DESCRIPTION
**Pull request by solid-yuriiuvarov**
Fixes #578 

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

**Review by illia-romanenko — CHANGES_REQUESTED** [review]
@solid-yuriiuvarov @solid-dimakoniaiev  you are only cleaning selectedProjectGroup, but you should also clean the list of project groups.

File: metrics/web/test/dashboard/presentation/state/project_metrics_notifier_test.dart [hunk 1]
```diff
@@ -669,6 +669,34 @@ void main() {
         );
       },
     );
+
+    test(
+      ".selectProjectGroup() resets project group dropdown items and selected project group to null if project groups are null",
```
Comment (solid-vladlaba) [inline metrics/web/test/dashboard/presentation/state/project_metrics_notifier_test.dart:674]:
```suggestion
      ".setProjectGroups() resets project group dropdown items and selected project group to null if project groups are null",
```
And seems like this test case lacks formatting.

**Review by solid-vladlaba — CHANGES_REQUESTED** [review]
Well done, @solid-yuriiuvarov @solid-dimakoniaiev! I've noticed one small mistake in a method name in tests. But overall everything is good!

**Review by illia-romanenko — APPROVED** [review]
LGTM - thanks @solid-yuriiuvarov @solid-dimakoniaiev !

**Pull request merged**